gherkin/python/test/gherkin_in_markdown_token_matcher_test.py

```diff
@@ -0,0 +1,232 @@
+# coding=utf-8
+
+import pytest
+from gherkin.token import Token
+from gherkin.token_matcher import GherkinInMarkdownTokenMatcher
+from gherkin.gherkin_line import GherkinLine
+location = { 'line': 1, 'column': 1 }
+
+def test_it_matches_FeatureLine():
+    tm = GherkinInMarkdownTokenMatcher('en')
+    line = GherkinLine('''## Feature: hello''',location['line'])
+    token = Token(gherkin_line=line, location=location)
+    assert tm.match_FeatureLine(token)
+    assert token.matched_type == 'FeatureLine'
+    assert token.matched_keyword == 'Feature'
+    assert token.matched_text == 'hello'
+
+def test_it_matches_FeatureLine_in_French():
+    tm = GherkinInMarkdownTokenMatcher('fr')
+    line = GherkinLine(u'''## Fonctionnalité: hello''',location['line'])
+    token = Token(gherkin_line=line, location=location)
+    assert tm.match_FeatureLine(token)
+    assert token.matched_type == 'FeatureLine'
+    assert token.matched_keyword == u'Fonctionnalité'
```

Comment:
I have resolved test failures in python 2.x by expliticly using unicode strings within the test.

Is this suitable/valid use or is it producing a false negative?

Reply:
I'm not sure. What is Pythons default encoding? What did change between 2/3?

Reply:
@mpkorstanje 
String encoding was a key change between 2 & 3.

https://stackoverflow.com/questions/3547534/what-encoding-do-normal-python-strings-use

> In Python 2: Normal strings (Python 2.x str) don't have an encoding: they are raw data.
>
> In Python 3: These are called "bytes" which is an accurate description, as they are simply sequences of bytes, which can be text encoded in any encoding (several are common!) or non-textual data altogether.
>
> For representing text, you want unicode strings, not byte strings. By "unicode strings", I mean unicode instances in Python 2 and str instances in Python 3. Unicode strings are sequences of unicode codepoints represented abstractly without an encoding; this is well-suited for representing text.

https://portingguide.readthedocs.io/en/latest/strings.html

My interpretation is:
* Strings `'something'` are unicode by default in Python 3 - hence no need to explicitly mark the keyword as unicode.
* The non-ASCII character `é` is not encoded correctly without explicit unicode conversion in Python 2.

Therefore I think this resolution is valid and not a false negative.

Do you agree with my understanding?

Reply:
Yeah that sounds reasonable. Thoug then I wonder what the encoding is of the source file in Python 2.

Reply:
[token_scanner](https://github.com/cucumber/common/blob/dd87c6d272e41148ec61cb75ea22625963251b30/gherkin/python/gherkin/token_scanner.py#L22) reads in the file and specifies UTF-8 - so that adds further weight.

I'm happy this shows that my change is valid rather than a false negative.

Reply:
No problems with the change. I'm not a python expert. 

I wouldn't even be opposed to dropping V2 support. Maybe on a separate PR thoug.

Reply:
Lol, wouldn't call myself a python expert either.  

I'll avoid commenting either way regarding version support.  I can live without V2 support.

Is there anything else you think is required before this change can be merged?
I think I've covered cases equally comprehensively as the javascript implementation.

Reply:
I'll try and have a look today. I would like to get this merged so I can split out Gherkin to it's own repository.